### ISSUE TITLE
pretraining a supervised deeplearner gives an error

### ISSUE THREAD

**Issue by peymanrah** [h2o/h2o#1343]
in Rstudio, if you use an autoencoder with the exact same config parameters as below and then use the autoencoder as a pretraining for a deep learning classification, it will through an error of mismatch between weights (the length of weights). Below are the code I used and the error that I keep receiving. 

I first use the autoencoder with below configuration, which works fine. 

AE_model_0 <- h2o.deeplearning(x = features_no_categorical_data,
                               training_frame = pre_training_hf,
                               model_id = "AE_model_0",
                               autoencoder = TRUE,
                               ignore_const_cols = TRUE,
                               seed = 42,
                               hidden = c(200, 100, 200), 
                               epochs = 100,
                               activation = "Tanh")

# then, I use a classification with an autoencoder as a pretrainer for weights. 

training_hf = as.h2o(mds %>% filter(usage == 'training'))

training_hf[,430] <- as.factor(training_hf[,430])  # for classification the label column (430) must be a factor.

DNN_model0 <- h2o.deeplearning(y = "v430",
                              x = features_no_categorical_data,
                              training_frame = training_hf,
                              pretrained_autoencoder  = "AE_model_0",
                              balance_classes = TRUE,
                              ignore_const_cols = TRUE,
                              seed = 42,
                              hidden = c(200, 100, 200), 
                              epochs = 100,
                              activation = "Tanh")

The below is the error I receive (I dont have any categorical variables in my features. I also enabled ignored_const_cols and I still receive the same error):

java.lang.IllegalArgumentException: Mismatch between weights in pretrained model and this model: cols in layer 0: 5716 vs 7019. Enable ignored_const_cols for both models and/or check categorical levels for consistency.

java.lang.IllegalArgumentException: Mismatch between weights in pretrained model and this model: cols in layer 0: 5716 vs 7019. Enable ignored_const_cols for both models and/or check categorical levels for consistency.
	at hex.deeplearning.DeepLearningModelInfo.initializeFromPretrainedModel(DeepLearningModelInfo.java:462)
	at hex.deeplearning.DeepLearning$DeepLearningDriver.buildModel(DeepLearning.java:238)
	at hex.deeplearning.DeepLearning$DeepLearningDriver.computeImpl(DeepLearning.java:216)
	at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:173)
	at hex.deeplearning.DeepLearning$DeepLearningDriver.compute2(DeepLearning.java:209)
	at water.H2O$H2OCountedCompleter.compute(H2O.java:1349)
	at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
	at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
	at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
	at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
	at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)

Error: java.lang.IllegalArgumentException: Mismatch between weights in pretrained model and this model: cols in layer 0: 5716 vs 7019. Enable ignored_const_cols for both models and/or check categorical levels for consistency.
In addition: Warning message:
In .h2o.startModelJob(algo, params, h2oRestApiVersion) :
  Dropping bad and constant columns: [v389, v390, v388].

**Comment by i110** [comment]
@peymanrah Hi, this repository is about H2O **web server** project, not for machine learning. [This](https://github.com/h2oai) might be  the one you're looking for.
